extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/glyptodon/guacamole/auth/jdbc/user/UserService.java

```diff
@@ -343,6 +348,41 @@ public ModeledUser retrieveUser(Credentials credentials)
         }
 
         // Return now-authenticated user
+        return user.getCurrentUser();
+
+    }
+
+    /**
+     * Retrieves the user corresponding to the given AuthenticatedUser from the
+     * database. If no such user exists, a placeholder entry will be created
+     * first.
+     *
+     * @param authenticatedUser
+     *     The AuthenticatedUser to retrieve the corresponding ModeledUser of.
+     *
+     * @return
+     *     The ModeledUser which corresponds to the given AuthenticatedUser,
+     *     created first if necessary.
+     */
+    public ModeledUser retrieveUser(org.glyptodon.guacamole.net.auth.AuthenticatedUser authenticatedUser) {
+
+        // If we already queried this user, return that rather than querying again
+        if (authenticatedUser instanceof AuthenticatedUser)
+            return ((AuthenticatedUser) authenticatedUser).getUser();
+
+        // Get username
+        String username = authenticatedUser.getIdentifier();
+
+        // Retrieve corresponding user model, if such a user exists
+        UserModel userModel = userMapper.selectOne(username);
+        if (userModel == null)
+            return null;
```

Comment:
Oops - it's not and it shouldn't. The placeholder entry idea was from a past iteration of the whole design of GUAC-586 - it was scrapped in recent development prior to this PR.

I'll fix the comment.